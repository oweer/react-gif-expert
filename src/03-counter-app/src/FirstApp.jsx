import PropTypes from 'prop-types';

export const FirstApp = ({ 
  name = 'Belly', 
  title = 'No hay título', 
  subTitle = 'No hay subtítulo' 
}) => {
  return (
    <>
    <h1 data-testid="test-title">{title}</h1>
    <p>{subTitle}</p>
      <p>{name}</p>
      
    </>
  );
};

