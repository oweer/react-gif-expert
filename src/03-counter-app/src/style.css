html, body {
    background-color: #21232A;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 10px;
}
button{
    padding: 5px;
    font-size: 1.2rem;
    margin-right: 5px;
}