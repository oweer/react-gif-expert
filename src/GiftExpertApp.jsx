import { useState } from "react";
import { AddCategory, GiftGrid } from "./components";


export const GiftExpertApp = () => {

    const [categories, setCategories] = useState([ 'Twilight']);

    const onAddCategory= ( newCategory )=>{

      if( categories.includes(newCategory)) return;
      setCategories( cat => [newCategory,...categories]);
      
    };

  return (
    <> 
        <h1>GiftExpertApp</h1>

      <AddCategory 
      onNewCategory={ (event) => onAddCategory(event)}
      />

          {categories.map( category => 
                (
                <GiftGrid 
                key={category} 
                category = {category}/>
            ))
          } 


    </>
  )
}
