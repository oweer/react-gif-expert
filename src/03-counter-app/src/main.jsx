import React from 'react';
import ReactDOM from 'react-dom/client';
//import {HelloWorldApp} from './HelloWorldApp';
import {FirstApp} from './FirstApp';
import {CounterApp} from './CounterApp';
import './style.css';


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
      {/* <HelloWorldApp /> */}
       <CounterApp value={100} />
      {/* <FirstApp title='Hola'/> */}
    </React.StrictMode>
);
