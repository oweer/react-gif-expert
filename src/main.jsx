import { StrictMode } from 'react';
import { createRoot } from 'react-dom/client';
import { GiftExpertApp } from './GiftExpertApp';
import './style.css';


createRoot(document.getElementById('root')).render(
  <StrictMode>
    <GiftExpertApp />
  </StrictMode>,
)
