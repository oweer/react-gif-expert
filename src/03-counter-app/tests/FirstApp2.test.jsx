/**
 * @jest-environment jsdom
 */

import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas de </FirstApp>', ()=>{

    const title = 'Hola, soy Goku';
    const subTitle = 'Hola, soy un subtitle';

  test('Debe hacer match con el snapchot', ()=>{
    const {container} = render (<FirstApp title={title} />);
    expect( container).toMatchSnapshot();

  })

  test('Debe de mostrar el mensaje "Hola soy goku"', ()=>{
    
    
    render (<FirstApp title={title} />)
    // screen.debug();
    expect( screen.getByText(title)).toBeTruthy();

  })

  test('Debe mostrar el titulo en un h1', ()=>{
    render (<FirstApp title={title} />);
    expect(screen.getByRole('heading', {level:1}).innerHTML).toContain(title)
    
  })

  test('Debe mostrar el subtitulo enviado por props', ()=>{
    render( 
      <FirstApp title={title} subTitle={subTitle} 
      /> );
      expect(screen.getAllByText(subTitle).length).toBe(1);

  })
    
});