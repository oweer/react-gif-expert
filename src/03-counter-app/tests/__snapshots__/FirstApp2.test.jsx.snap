// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de </FirstApp> Debe hacer match con el snapchot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Belly
  </p>
</div>
`;
